rust_src/src/eval_macros.rs

```diff
@@ -67,3 +67,45 @@ macro_rules! error {
 macro_rules! def_lisp_sym {
     ($name:expr, $value:expr) => {};
 }
+
+/// Macros we use to define forwarded Lisp variables.
+/// These are used in the syms_of_FILENAME functions.
+///
+/// An ordinary (not in buffer_defaults, per-buffer, or per-keyboard)
+/// lisp variable is actually a field in `struct emacs_globals'.
+///
+/// In the C code, the field's name begins with "f_", which is a
+/// convention enforced by these macros.  Each such global has a
+/// corresponding #define in globals.h; the plain name should be used
+/// in the C code.
+///
+/// E.g., the global "cons_cells_consed" is declared as "int
+/// f_cons_cells_consed" in globals.h, but there is a define:
+///
+///    #define cons_cells_consed globals.f_cons_cells_consed
+///
+/// All C code uses the `cons_cells_consed' name.
+///
+/// As the Rust macro system has identifier hygine, the Rust code's
+/// version of the struct emacs_globals does not include the f_ prefix
+/// on the field names, and Rust code accesses the fields directly,
+/// rather than through a macro.
+///
+/// This is all done this way to support indirection for
+/// multi-threaded Emacs.
+#[macro_export]
+macro_rules! defvar_lisp {
```

Comment:
I think this version defvar_lisp has a bug.
I fixed it in https://github.com/emacs-ng/emacs-ng/pull/35/commits/6571919fb2b9c59539a888544d5f7bbd02fd3b43#diff-1315856f6f17c5ca7f91bfbfac358d98fa56279159b633ffff1ed82bdf2f05deR104

Reply:
Addressed, thank you for the heads up.